### PR TITLE
Adding sphinxcontrib-getthecode

### PR DESCRIPTION
**Pull request by FabriceSalvaire**
@conda-forge/help-python 

**uploaded**: https://anaconda.org/FabriceSALVAIRE/sphinxcontrib-getthecode

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by FabriceSalvaire** [comment]
@conda-forge/staged-recipes This is ready for review.

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+requirements:
+  build:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - sphinx >=3.0
+# build:
+#   - wheel
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:36]: Should this be dropped?

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  imports:
+
+about:
+  home: https://github.com/FabriceSalvaire/sphinx-getthecode
+  # https://www.gnu.org/licenses/gpl-3.0.en.html
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:45]: Maybe add this as `license_url`

```suggestion
  license_url: https://www.gnu.org/licenses/gpl-3.0.en.html
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  license_family: GPL
+  license_file: LICENSE.txt
+  summary: 'sphinxcontrib-getthecode is Sphinx add-ons to generate documentation for Python projects'
+  # description:
+  doc_url: https://github.com/FabriceSalvaire/sphinx-getthecode
+  dev_url: https://github.com/FabriceSalvaire/sphinx-getthecode
+
+extra:
+  # maintainers:
+  #   - FabriceSalvaire
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:56]: Should this be dropped?

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  skip: True  # [py<34]
+  script: "{{ PYTHON }} -m pip install . -vv"
+  has_prefix_files:
+  always_include_files:
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:28]: No need for empty fields.

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+{% set name = "sphinxcontrib-getthecode" %}
+{% set version = "1.2.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/FabriceSalvaire/sphinx-getthecode/archive/v{{ version }}.tar.gz
+  # run openssl sha256 <file name>
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:10]: Drop example comments

```suggestion
  # run openssl sha256 <file name>
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  sha256: 9e84b57b36f07d6dbe157521cc9033cb491142af7dc45b277bf608f8e5b84eda
+  # git_url: https://github.com/FabriceSalvaire/sphinx-getthecode
+  # git_rev: ... # "v{{ version }}"
+  # git_depth: 1 # (Defaults to -1/not shallow)
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:14]: Drop example comments

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+build:
+  # noarch: python
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:17]: Is it pure Python? If so, can we make it `noarch`?

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:21]: Drop example comments

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  skip: True  # [py<34]
+  script: "{{ PYTHON }} -m pip install . -vv"
+  has_prefix_files:
+  always_include_files:
+
+requirements:
+  build:
+  host:
+    - python
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:30]: For using `noarch`, constrain Python like so

```suggestion
    - python >=3.4
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    - pip
+  run:
+    - python
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:33]: Same here

```suggestion
    - python >=3.4
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  # git_rev: ... # "v{{ version }}"
+  # git_depth: 1 # (Defaults to -1/not shallow)
+
+build:
+  # noarch: python
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  skip: True  # [py<34]
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:22]: Then drop this

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+  has_prefix_files:
+  always_include_files:
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:25]: Drop empty fields

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    - pip
+  run:
+    - python
+    - sphinx >=3.0
+# build:
+#   - wheel
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:40]: Drop example comments

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+  imports:
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:41]: Can you please add some `import`s here for testing?

File: recipes/sphinxcontrib-getthecode/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+
+about:
+  home: https://github.com/FabriceSalvaire/sphinx-getthecode
+  # https://www.gnu.org/licenses/gpl-3.0.en.html
+  license: GPL-3.0-or-later
+  license_family: GPL
+  license_file: LICENSE.txt
+  summary: 'sphinxcontrib-getthecode is Sphinx add-ons to generate documentation for Python projects'
+  # description:
```
Comment (jakirkham) [inline recipes/sphinxcontrib-getthecode/meta.yaml:50]: Drop unused commented field.

```suggestion
```

File: recipes/sphinxcontrib-getthecode/meta.yaml (hunk 5 above)
Comment (FabriceSalvaire) [reply recipes/sphinxcontrib-getthecode/meta.yaml:10]: just a comment

File: recipes/sphinxcontrib-getthecode/meta.yaml (hunk 14 above)
Comment (FabriceSalvaire) [reply recipes/sphinxcontrib-getthecode/meta.yaml:28]: test are not implemented

**Comment by FabriceSalvaire** [comment]
@conda-forge/staged-recipes This is ready for review. And a dependency of conda-forge/pyspice-feedstock

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**